champ/node/lib.rs

```diff
@@ -55,18 +55,23 @@ pub async fn run() -> Result<()> {
     debug!("initializing blockpool");
     let mut blockpool = Blockpool::new();
 
-    debug!("initializing wallet manager");
+    debug!("initializing wallet manager step 1/2");
     let wallet_manager = WalletManager::new(config.read().await.wallets.clone());
-    let wallet_manager = RwLock::new(wallet_manager);
+    let wallet_lock = RwLock::new(wallet_manager);
 
     debug!("initializing champ state");
     let state = ChampState::new(ChampStateArgs {
         db,
         config,
-        wallet_manager,
+        //wallet_manager,
```

Comment:
we need the wallet manager in the state :( Can also check this out I know rust's compiler can be a bi*ch with thread safety. Doesn't the `wallet_lock` already work here?